src/civetweb.c

```diff
@@ -9449,10 +9481,22 @@ handle_request(struct mg_connection *conn)
 		}
 
 		/* 6. authorization check */
-		if (is_put_or_delete_request && !is_script_resource
-		    && !is_callback_resource) {
-/* 6.1. this request is a PUT/DELETE to a real file */
-/* 6.1.1. thus, the server must have real files */
+		/* 6.1. a custom authorization handler is installed */
+		if (get_request_handler(conn,
+		                        AUTH_HANDLER,
+		                        &auth_handler,
+		                        NULL,
+		                        NULL,
+		                        NULL,
+		                        NULL,
+		                        &auth_callback_data)) {
+			if (!auth_handler(conn, auth_callback_data)) {
```

Comment:
```
/* 5.2. check if the request will be handled by a callback */
if (get_request_handler(...)) {
    /* 5.2.1. A callback will handle this request. All requests handled
     * by a callback have to be considered as requests to a script
     * resource. */

    callback_resource = 1;
    ...
}
```
So every resource with an auth handler will be considered a script resource.

I think this will cause a problem if you set a auth handler for a static file.
Did you test this case?